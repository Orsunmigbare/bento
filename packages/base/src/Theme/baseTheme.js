import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
