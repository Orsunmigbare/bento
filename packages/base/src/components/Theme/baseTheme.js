export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
