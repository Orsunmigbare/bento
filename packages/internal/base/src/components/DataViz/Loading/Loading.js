import React from "react"
import PropTypes from "prop-types"
import styled, { keyframes } from "styled-components"
import { color, spacing, themify, ThemeComponent } from "@staccx/theme"

const DefaultLoading = ({ className, variant, ...rest }) => (
  <SignalSpinner className={className} variant={variant} {...rest} />
)

export const COMPONENT_LOADING = "COMPONENT_LOADING"

const Loading = ({ className, variant, ...rest }) => (
  <ThemeComponent
    tagName={COMPONENT_LOADING}
    fallback={DefaultLoading}
    {...rest}
  />
)

const spinnerBubble1 = keyframes`
  0% {
    color: ${color.primary};
    transform: translate(-50%, -50%) scale(1);
    opacity: 1;
  }
  100% {
    color:  ${color.secondary};
    transform: translate(-50%, -50%) scale(4.8);
    opacity: 0;
  }
`

const spinnerBubble2 = keyframes`
  0% {
    color: ${color.secondary};
    transform: translate(-50%, -50%) scale(1);
    opacity: 1;
  }
  100% {
    color: ${color.primary};
    transform: translate(-50%, -50%) scale(4.8);
    opacity: 0;
  }
`

export const LOADING = "LOADING"
const SignalSpinner = styled.div`
  position: relative;
  display: block;
  width: ${spacing.large};
  height: ${spacing.large};
  color: ${color.primary};

  &::before,
  &::after {
    content: "";
    position: absolute;
    top: 50%;
    left: 50%;
    width: 10px;
    height: 10px;
    opacity: 0;
    box-shadow: 0 0 0 1px;
    border-radius: 50%;
    transform: translate(-50%, -50%) scale(1);
    transform-origin: center;
  }

  &::before {
    animation: ${spinnerBubble1} 2.6s ease-out 1.3s infinite;
  }

  &::after {
    animation: ${spinnerBubble2} 2.6s ease-out 0s infinite;
  }
  ${themify(LOADING)};
`

Loading.propTypes = { className: PropTypes.string, variant: PropTypes.string }

export default Loading
