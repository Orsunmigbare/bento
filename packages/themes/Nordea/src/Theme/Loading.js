import React from "react"
import styled, { keyframes } from "styled-components"
import { Loading, theming } from "@staccx/base"

const NordeaLoader = () => (
  <svg xmlns="http://www.w3.org/2000/Svg" viewBox="0 0 292 292" width="72">
    <g fill="#0000A0" fillRule="evenodd">
      <Path d="M11.2,160.2 C5.2,160.2 0.3,155.3 0.3,149.3 L0.3,134.7 C0.3,128.7 5.2,123.8 11.2,123.8 C17.2,123.8 22.1,128.7 22.1,134.7 L22.1,149.3 C22.1,155.4 17.2,160.2 11.2,160.2 Z" />
      <Path d="M54.9,162.1 C44.9,162.1 36.7,154 36.7,143.9 L36.7,89.4 C36.7,79.4 44.8,71.2 54.9,71.2 C64.9,71.2 73.1,79.3 73.1,89.4 L73.1,143.9 C73,153.9 64.9,162.1 54.9,162.1 Z" />
      <Path d="M113.9,203.9 C101.4,203.9 91.2,193.7 91.2,181.2 L91.2,23.1 C91.2,10.6 101.4,0.4 113.9,0.4 C126.4,0.4 136.6,10.6 136.6,23.1 L136.6,181.2 C136.6,193.7 126.5,203.9 113.9,203.9 Z" />
      <Path d="M177.5,283.8 C165,283.8 154.8,273.6 154.8,261.1 L154.8,103 C154.8,90.5 165,80.3 177.5,80.3 C190,80.3 200.2,90.5 200.2,103 L200.2,261.1 C200.2,273.6 190.1,283.8 177.5,283.8 Z" />
      <Path d="M236.6,212.9 C226.6,212.9 218.4,204.8 218.4,194.7 L218.4,140.2 C218.4,130.2 226.5,122 236.6,122 C246.6,122 254.8,130.1 254.8,140.2 L254.8,194.7 C254.7,204.8 246.6,212.9 236.6,212.9 Z" />
      <Path
        fillRule="nonzero"
        d="M280.2,160.2 C274.2,160.2 269.3,155.3 269.3,149.3 L269.3,134.7 C269.3,128.7 274.2,123.8 280.2,123.8 C286.2,123.8 291.1,128.7 291.1,134.7 L291.1,149.3 C291.1,155.4 286.2,160.2 280.2,160.2 Z"
      />
    </g>
  </svg>
)

export const ScaleInY = keyframes`
  0%, 60% {
    transform: scaleY(0) scaleX(0.3);
  }

  20%,
  40%{
    transform: scaleY(1) scaleX(1);
  }
`

const Path = styled.path`
  animation: ${ScaleInY} 3s cubic-bezier(0.59, 0.24, 0.33, 1.1) forwards
    infinite;
  transform: scaleY(0) scaleX(0.3);
  transform-origin: center;
  &:nth-child(1) {
    transform-origin: 5% center;
  }
  &:nth-child(2) {
    transform-origin: 20% center;
    animation-delay: 0.1s;
  }
  &:nth-child(3) {
    transform-origin: 40% center;
    animation-delay: 0.2s;
  }
  &:nth-child(4) {
    transform-origin: 55% center;
    animation-delay: 0.3s;
  }
  &:nth-child(5) {
    transform-origin: 75% center;
    animation-delay: 0.4s;
  }
  &:nth-child(6) {
    transform-origin: 90% center;
    animation-delay: 0.5s;
  }
`

export const LoadingStyle = theming.createVariants(
  { [theming.VARIANT_DEFAULT]: NordeaLoader },
  Loading.themeProps.component
)
