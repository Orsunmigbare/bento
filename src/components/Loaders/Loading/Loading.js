import React from "react"
import PropTypes from "prop-types"
import styled, { keyframes } from "styled-components"
import { themePropTypes } from "../../../constants/themeContants"
import ThemeComponent from "../../Theme/ThemeComponent"
import { applyVariants, color, spacing } from "../../../theming"
import useTimer from "../../../hooks/useTimer"

const DefaultLoading = ({ className, variant, ...rest }) => (
  <SignalSpinner className={className} variant={variant} {...rest} />
)

export const COMPONENT_LOADING = "COMPONENT_LOADING"

const themeProps = {
  component: {
    name: "COMPONENT_LOADING",
    description: "Component for loading",
    type: themePropTypes.component
  },
  loading: {
    name: "LOADING",
    description: "Loading style",
    type: themePropTypes.style
  }
}

const Loading = ({ className, idleTime, ...rest }) => {
  const isReady = useTimer(idleTime)

  // Wait before rendering
  if (!isReady) {
    return null
  }
  return (
    <ThemeComponent
      tagName={themeProps.component.name}
      fallback={DefaultLoading}
      {...rest}
    />
  )
}

const spinnerBubble1 = keyframes`
  0% {
    color: ${color.primary};
    transform: translate(-50%, -50%) scale(1);
    opacity: 1;
  }
  100% {
    color:  ${color.secondary};
    transform: translate(-50%, -50%) scale(4.8);
    opacity: 0;
  }
`

const spinnerBubble2 = keyframes`
  0% {
    color: ${color.secondary};
    transform: translate(-50%, -50%) scale(1);
    opacity: 1;
  }
  100% {
    color: ${color.primary};
    transform: translate(-50%, -50%) scale(4.8);
    opacity: 0;
  }
`

const SignalSpinner = styled.div`
  position: relative;
  display: block;
  width: ${spacing.large};
  height: ${spacing.large};
  color: ${color.primary};

  &::before,
  &::after {
    content: "";
    position: absolute;
    top: 50%;
    left: 50%;
    width: 10px;
    height: 10px;
    opacity: 0;
    box-shadow: 0 0 0 1px;
    border-radius: 50%;
    transform: translate(-50%, -50%) scale(1);
    transform-origin: center;
  }

  &::before {
    animation: ${spinnerBubble1} 2.6s ease-out 1.3s infinite;
  }

  &::after {
    animation: ${spinnerBubble2} 2.6s ease-out 0s infinite;
  }
  ${applyVariants(themeProps.loading.name)};
`

Loading.propTypes = {
  className: PropTypes.string,
  /**
   * Allows different styles
   */
  variant: PropTypes.string
}
Loading.themeProps = themeProps

export default Loading
