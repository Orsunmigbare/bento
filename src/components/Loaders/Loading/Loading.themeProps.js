import { themePropTypes } from "../../../theming"

const themeProps = {
  component: {
    name: "COMPONENT_LOADING",
    description: "Component for loading",
    type: themePropTypes.component
  },
  loading: {
    name: "LOADING",
    description: "Loading style",
    type: themePropTypes.style
  }
}

export default themeProps
