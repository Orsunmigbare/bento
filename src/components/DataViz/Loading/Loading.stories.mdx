import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Loading from "./Loading.js"

<Meta title="Components|DataViz/Loading" component={Loading}/>

# Stories for Loading


## Loading
<Preview>
  <Story name="0%">
   <Loading/>
  </Story>
</Preview>

<Props of={Loading}/>
