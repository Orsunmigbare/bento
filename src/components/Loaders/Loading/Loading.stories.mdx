import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Loading from "./Loading.js"

<Meta title="Components|Loaders/Loading" component={Loading}/>

# Loading
Component to illustrate that the application is loading. Often used while loading api data and other async operations
# import

```javascript
Import {Loading} from "@staccx/bento"
```

## Loading
<Preview>
  <Story name="Loading">
   <Loading/>
  </Story>
</Preview>

# Props for Loading

<Props of={Loading}/>
