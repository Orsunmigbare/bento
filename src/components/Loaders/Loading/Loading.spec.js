import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import Loading from "./Loading"

describe("Loading", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Loading />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("className", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Loading className="className" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("isReady", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Loading isReady />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
