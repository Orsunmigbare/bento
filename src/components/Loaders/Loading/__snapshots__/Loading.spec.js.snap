// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Snapshots No props 1`] = `null`;

exports[`Loading Snapshots className 1`] = `null`;

exports[`Loading Snapshots isReady 1`] = `null`;
